src/main/java/seedu/address/model/event/Event.java

```diff
@@ -233,6 +234,59 @@ public void removePerson(Person person, String role) {
         roleSet.remove(person);
 
     }
+
+    /**
+     * Removes person from all roles if they are present in that role
+     * @param personToRemove Person that will be removed
+     */
+    public void removePerson(Person personToRemove) {
+        requireNonNull(personToRemove);
+        // check each set and see it contains the person to remove
+        if (attendees.contains(personToRemove)) {
+            attendees.remove(personToRemove);
+        }
+
+        if (vendors.contains(personToRemove)) {
+            vendors.remove(personToRemove);
+        }
+
+        if (sponsors.contains(personToRemove)) {
+            sponsors.remove(personToRemove);
+        }
+
+        if (volunteers.contains(personToRemove)) {
+            volunteers.remove(personToRemove);
+        }
+    }
+
+    /**
+     * Edits person in all roles if they are present in that role
+     * @param personToEdit Person that will be removed
+     * @param editedPerson Person that will replace personToEdit
+     */
+    public void editPerson(Person personToEdit, Person editedPerson) {
+        requireAllNonNull(personToEdit, editedPerson);
+        // check each set and see it contains the person to edit
+        if (attendees.contains(personToEdit)) {
+            attendees.remove(personToEdit);
+            attendees.add(editedPerson);
+        }
+
+        if (vendors.contains(personToEdit)) {
+            vendors.remove(personToEdit);
+            vendors.add(editedPerson);
+        }
+
+        if (sponsors.contains(personToEdit)) {
+            sponsors.remove(personToEdit);
+            sponsors.add(editedPerson);
+        }
+
+        if (volunteers.contains(personToEdit)) {
+            volunteers.remove(personToEdit);
+            volunteers.add(editedPerson);
+        }
```

Comment:
Can't think of anyway better to do this rn but this does seem abit funny caz we have to check every set and add / remove, maybe we can think of something better next time

Reply:
Perhaps we could abstract out this removal and addition into a method called editPersonInRoleSet to make it more readable and less repetitive, but otherwise the implementation seems sound